orca-clouddriver/src/test/java/com/netflix/spinnaker/orca/clouddriver/tasks/providers/cf/CloudFoundryCreateServiceKeyTaskTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.cf;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.KatoService;
+import com.netflix.spinnaker.orca.clouddriver.model.TaskId;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.junit.jupiter.api.Test;
+import rx.Observable;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType.PIPELINE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.matches;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class CloudFoundryCreateServiceKeyTaskTest {
+  @Test
+  void shouldMakeRequestToKatoToCreateServiceKey() {
+    String type = "createServiceKey";
+    KatoService kato = mock(KatoService.class);
+    String cloudProvider = "my-cloud";
+    String credentials = "cf-foundation";
+    String region = "org > space";
+    String serviceInstanceName = "service-instance";
+    String serviceKeyName = "service-key";
+    TaskId taskId = new TaskId("kato-task-id");
+    Map<String, Object> context = new HashMap<>();
```

Comment:
Using `ImmutableMap` below. Can we use that here too for consistency? Can we remove some of these local variables and inline them to shorten up the test?

Reply:
The map above is modified during the course of the test so it cannot be an ImmutableMap.  Would you prefer us to convert the ImmutableMap into a HashMap?